### PR TITLE
Fixed parenthesis enclosed expression issue  

### LINKED ISSUE #442: `(.)` returns a misshapen node error
### Description
Running `(.)` returns a misshapen node error.

### Steps to Reproduce
1. evaluate `(.)`

**Expected behavior:**
Returns whatever value bind to `.`

**Actual behavior:**
```
INFO[0000] misshapen node AST: (
  @choice: [0, 6],
  cdelim: ('': 2‣")"),
  identpattern: (IDENT: ('': 1‣.)),
  odelim: ('': 0‣"("),
)
```

### Your Environments
```sh
$ go version
go version go1.14.2 darwin/amd64

```

### PR DESCRIPTION
**Pull request by ericzhang6222**
Fixes #442 .

Changes proposed in this pull request:
- Will pasre `(a)` in `let a = 5;(a)` to expression.

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: syntax/arrai.wbnf [hunk 1]
```diff
@@ -80,7 +81,7 @@ SEQ_COMMENT -> "," C*;
   | C* odelim="[" C* array=(%!sparse_sequence(top)?) C* cdelim="]" C*
   | C* odelim="<<" C* bytes=(item=(STR|NUM|CHAR|IDENT|"("top")"):SEQ_COMMENT,?) C* cdelim=">>" C*
   | C* odelim="(" tuple=(pairs=(extra | ((rec="rec"? name| name?) ":" v=top)):SEQ_COMMENT,?) cdelim=")" C*
-  | C* odelim="(" identpattern=IDENT cdelim=")" C*
+  | C* odelim="(" identpattern=IDENT_NODOT cdelim=")" C*
```
Comment (marcelocantos) [inline syntax/arrai.wbnf:84]: This won't fix `(x)`, which should also be treated as an expression, not an `IdentExpr`. I suspect this line needs to disappear entirely.

Comment (ericzhang6222) [reply syntax/arrai.wbnf:84]: But I got an example in let_expr_test.go:
```
AssertCodeErrors(t, `let a = 5; let (a) = 1 + 2; a`, "")
```
It means `let a = 5; let (a) = 1 + 2; a` will raise following errors:
```
$ arrai e 'let a = 5; let (a) = 1 + 2; a'
INFO[0000] a is redefined differently 5 vs 3

.:1:12:
let a = 5; let (a) = 1 + 2; a 
```
in this case, `(a)` is parsed to an `IdentExpr` and it is expected behavior.

Comment (marcelocantos) [reply syntax/arrai.wbnf:84]: The following code should work:
```arrai
let a = 5; (a)
```

File: syntax/expr_let_test.go [hunk 2]
```diff
@@ -11,6 +11,14 @@ func TestExprLet(t *testing.T) {
...
 	AssertCodesEvalToSameValue(t, `(x: 1, y: 2)`, `let x = 1; let y = 2; (:x, :y)`)
 	AssertCodesEvalToSameValue(t, `(x: 1, y: 2)`, `let x = 1; (:x, y: 2)`)
+
+	AssertCodesEvalToSameValue(t, `1`, `let a = 1;(a)`) // (a) is an expression
+	AssertCodesEvalToSameValue(t, `4`, `let x = 4; let (x) = 4;x`)
+	AssertCodesEvalToSameValue(t, `4`, `let x = 4; let (x) = (4);x`)
+	AssertCodesEvalToSameValue(t, `4`, `let x = 4; let (x) =(4 + 0);x`)
+	AssertCodesEvalToSameValue(t, `1`, `let [a,b] = [1,2];let (b) = (a + 1);a`)
+
+	AssertCodeErrors(t, `let (x) = 5;x`, "") // (x) should be parsed as an expression and fail because x isn't bound.
```
Comment (marcelocantos) [inline syntax/expr_let_test.go:21]: Also need a case for bound but has a different value.

**Pull request merged**